sdks/java/io/redis/src/main/java/org/apache/beam/sdk/io/redis/RedisIO.java

```diff
@@ -0,0 +1,294 @@
+import static com.google.common.base.Preconditions.checkState;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.annotations.VisibleForTesting;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.KvCoder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.io.BoundedSource;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.apache.beam.sdk.transforms.display.DisplayData;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PBegin;
+import org.apache.beam.sdk.values.PCollection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An IO to manipulate Redis key/value database.
+ *
+ * <h3>Reading Redis key/value pairs</h3>
+ *
+ * <p>RedisIO.Read provides a source which returns a bounded {@link PCollection} containing
+ * key/value pairs as {@code KV<String, String>}.
+ *
+ * <p>To configure a Redis source, you have to provide Redis server hostname and port number.
+ * Optionally, you can provide a key pattern (to filter the keys). The following example
+ * illustrates how to configure a source:
+ *
+ * <pre>{@code
+ *
+ *  pipeline.apply(RedisIO.read()
+ *    .withConnection(RedisConnection.create(Collections.singletonList("localhost:6379")))
+ *    .withKeyPattern("foo*")
+ *
+ * }</pre>
+ */
+public class RedisIO {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RedisIO.class);
+
+  public static Read read() {
+    return new  AutoValue_RedisIO_Read.Builder()
+        .setCommand(Read.Command.GET).setKeyPattern("*").build();
+  }
+
+  private RedisIO() {
+  }
+
+  /**
+   * A {@link PTransform} reading key/value pairs from a Redis database.
+   */
+  @AutoValue
+  public abstract static class Read extends PTransform<PBegin, PCollection<KV<String, String>>> {
+
+    /**
+     * The Redis commands related to read of key-value pairs.
+     */
+    public enum Command {
+      GET
+    }
+
+    @Nullable abstract RedisConnection connection();
+    @Nullable abstract Command command();
+    @Nullable abstract String keyPattern();
+    @Nullable abstract RedisService redisService();
+
+    abstract Builder builder();
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      @Nullable abstract Builder setConnection(RedisConnection connection);
+      @Nullable abstract Builder setCommand(Command command);
+      @Nullable abstract Builder setKeyPattern(String keyPattern);
+      @Nullable abstract Builder setRedisService(RedisService redisService);
+      abstract Read build();
+    }
+
+    /**
+     * Define the connection to the Redis server.
+     *
+     * @param connection The {@link RedisConnection}.
```

Comment:
Remove useless parts from all comments (those that don't add any information that isn't already obvious from the function signature).